core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -153,7 +153,12 @@ class GroupCoordinator(val brokerId: Int,
             // coordinator OR the group is in a transient unstable phase. Let the member retry
             // joining without the specified member id,
             responseCallback(joinError(memberId, Errors.UNKNOWN_MEMBER_ID))
-
+          case InitialRebalance =>
+            if (memberId != JoinGroupRequest.UNKNOWN_MEMBER_ID) {
+              responseCallback(joinError(memberId, Errors.UNKNOWN_MEMBER_ID))
```

Comment:
I think there is a rare scenario that the group has just been removed from the coordinator cache, then some existing consumer members just send their join groups immediately afterwards. In this case, we should still accept the request, in that "I forgot this group does not mean this group never exist".